docs/spec/v1beta2/gitrepositories.md

```diff
@@ -0,0 +1,812 @@
+type: Opaque
+data:
+  author1.asc: <BASE64>
+  author2.asc: <BASE64>
+```
+
+Exporting armored public keys (`.asc` files) using `gpg`, and generating a
+Secret:
+
+```sh
+# Export armored public keys
+gpg --export --armor 3CB12BA185C47B67 > author1.asc
+gpg --export --armor 6A7436E8790F8689 > author2.asc
+# Generate secret
+kubectl create secret generic pgp-public-keys \
+    --from-file=author1.asc \
+    --from-file=author2.asc \
+    -o yaml
+```
+
+### Ignore
+
+`.spec.ignore` is an optional field to specify rules in [the `.gitignore`
+pattern format](https://git-scm.com/docs/gitignore#_pattern_format). Paths
+matching the defined rules are excluded while archiving.
+
+When specified, `.spec.ignore` overrides the [default exclusion
+list](#default-exclusions), and may overrule the [`.sourceignore` file
+exclusions](#sourceignore-file). See [excluding files](#excluding-files)
+for more information.
+
+### Suspend
+
+`.spec.suspend` is an optional field to suspend the reconciliation of a
+GitRepository. When set to `true`, the controller will stop reconciling the
+GitRepository, and changes to the resource or in the Git repository will not
+result in a new Artifact. When the field is set to `false` or removed, it will
+resume.
+
+### Git implementation
+
+`.spec.gitImplementation` is an optional field to change the client library
+implementation used for Git operations (e.g. clone, checkout). The default
+value is `go-git`.
+
+Unless you need support for a specific Git wire protocol functionality not
+supported by the default implementation (as documented below), changing the
+implementation is generally not recommended as it can come with its own set of
+drawbacks. For example, not being able to make use of shallow clones forces the
+controller to fetch the whole Git history tree instead of a specific one,
+resulting in an increase of disk space and traffic usage.
+
+| Git Implementation | Shallow Clones | Git Submodules | V2 Protocol Support |
+|--------------------|----------------|----------------|---------------------|
+| `go-git`           | true           | true           | false               |
+| `libgit2`          | false          | false          | true                |
+
+Some Git providers like Azure DevOps _require_ the `libgit2` implementation, as
+their Git servers provide only support for the
+[v2 protocol](https://git-scm.com/docs/protocol-v2).
+
+#### Proxy support
+
+When a proxy is configured in the source-controller Pod through the appropriate
+environment variables, for example `HTTPS_PROXY`, `NO_PROXY`, etc. There may be
+some limitations in the proxy support based on the Git implementation.
+
+| Git Implementation | HTTP_PROXY | HTTPS_PROXY | NO_PROXY | Self-signed Certs |
+|--------------------|------------|-------------|----------|-------------------|
+| `go-git`           | true       | true        | true     | false             |n
+| `libgit2`          | false      | true        | true     | true              |
```

Comment:
`libgit2 1.3.0` seem to have support for `HTTP_PROXY`:

https://github.com/libgit2/libgit2/blob/v1.3.0/src/remote.c#L946-L951

```suggestion
| `go-git`           | true       | true        | true     | false             |
| `libgit2`          | true       | true        | true     | true              |
```